driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -267,7 +267,7 @@ OperationResult executeListDatabaseNames(final BsonDocument operation) {
     OperationResult executeListCollections(final BsonDocument operation) {
         MongoDatabase database = entities.getDatabase(operation.getString("object").getValue());
 
-        BsonDocument arguments = operation.getDocument("arguments");
+        BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
```

Comment:
This change was actually necessary to get the tests to pass.  The rest below I changed because this keeps happening.